src/Npgsql/ConnectorPool.cs

```diff
@@ -130,12 +130,14 @@ internal ConnectorPool(NpgsqlConnectionStringBuilder settings, string connString
 
             var connectionIdleLifetime = TimeSpan.FromSeconds(Settings.ConnectionIdleLifetime);
             _pruningSamplingInterval = TimeSpan.FromSeconds(Settings.ConnectionPruningInterval);
+            if (Settings.ConnectionPruningInterval == 0)
```

Comment:
Let's check arguments earlier, before assigning `settings` to `Settings`.

Reply:
Agreed, I'll move it into settings.

Reply:
Or do you mean just moving it up in this constructor? That'd be fairly pointless IMO.

Reply:
> Or do you mean just moving it up in this constructor?

This one. Not pointless because there is the same argument validation We don't validate connection string right now during its construction and this should be tracked in the different issue (in some cases we can't validate a string before assigning all properties due to dependencies between them).

Reply:
Ah I see what you mean now, that's fine. 

Validation, what about the 0 check, should I move that one to connection string or better tackle all validation issues at once?

Reply:
I think at once and after some discussion. As I said, there are relationships between properties (for example `MaxPoolSize` should be greater than `MinPoolSize` which prevents unordered assignments), so we should know how to do it better (maybe add a method to validate the string).

Reply:
Oh I understand the problem (it's solution is similar to aspnet core options validation, so indeed a second pass) though I wasn't sure where we landed on validation as much as early as possible or rather keep the checks grouped together. I'll move them to settings and leave it at that!